src/router.ts

```diff
@@ -2,77 +2,91 @@ import { createRouter, createWebHashHistory } from 'vue-router';
 import domains from '@/../snapshot-spaces/spaces/domains.json';
 import aliases from '@/../snapshot-spaces/spaces/aliases.json';
 import Home from '@/views/Home.vue';
-import Proposal from '@/views/Proposal.vue';
-import Create from '@/views/Create.vue';
+import SpaceProposal from '@/views/SpaceProposal.vue';
+import SpaceCreate from '@/views/SpaceCreate.vue';
 import Setup from '@/views/Setup.vue';
-import Settings from '@/views/Settings.vue';
+import SpaceSettings from '@/views/SpaceSettings.vue';
 import Explore from '@/views/Explore.vue';
 import Strategy from '@/views/Strategy.vue';
 import Playground from '@/views/Playground.vue';
 import Delegate from '@/views/Delegate.vue';
 import Timeline from '@/views/Timeline.vue';
 import Space from '@/views/Space.vue';
-import About from '@/views/About.vue';
+import SpaceAbout from '@/views/SpaceAbout.vue';
+import SpaceProposals from '@/views/SpaceProposals.vue';
 
 const domainName = window.location.hostname;
 
-const beforeEnter = (to: any, from, next) => {
-  if (aliases?.[to?.params?.key]) {
-    to.params.key = aliases[to.params.key];
-    return next(to);
-  }
-  next();
-};
-
 const routes: any[] = [
   {
     path: '/',
     name: 'home',
-    component: domains[domainName] ? Space : Home
+    component: Home,
+    beforeEnter: (to, from, next) => {
+      if (domains[domainName])
+        next({ name: 'spaceProposals', params: { key: domains[domainName] } });
```

Comment:
This is going to redirect vote.balancer.fi to vote.balance.fi/#/balancer?

Reply:
Yes, couldn't find another way to do it with nested routes

Reply:
Did you tried to make 2 routers, 1 for custom domains and a router for main site?